src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/MSBuildProject.NuGetPackages.cs

```diff
@@ -14,7 +16,84 @@ namespace Microsoft.DotNet.UpgradeAssistant.MSBuild
 {
     internal partial class MSBuildProject : INuGetReferences
     {
-        public INuGetReferences NuGetReferences => this;
+        public async ValueTask<INuGetReferences> GetNuGetReferencesAsync(CancellationToken token)
+        {
+            if (IsRestored)
+            {
+                return this;
+            }
+
+            await _restorer.RestorePackagesAsync(Context, this, token).ConfigureAwait(false);
+
+            return this;
+        }
+
+        public NugetPackageFormat PackageReferenceFormat
+        {
+            get
+            {
+                if (GetPackagesConfigPath() is not null)
+                {
+                    return NugetPackageFormat.PackageConfig;
+                }
+                else if (ProjectRoot.GetAllPackageReferences().ToList() is IEnumerable<Build.Construction.ProjectItemElement> list && list.Any())
+                {
+                    return NugetPackageFormat.PackageReference;
+                }
+                else
+                {
+                    return NugetPackageFormat.None;
+                }
+            }
+        }
+
+        private string? GetPackagesConfigPath() => FindFiles(ProjectItemType.Content, "packages.config").FirstOrDefault();
+
+        public IEnumerable<NuGetReference> PackageReferences
+        {
+            get
+            {
+                var packagesConfig = GetPackagesConfigPath();
+
+                if (packagesConfig is null)
+                {
+                    var packages = ProjectRoot.GetAllPackageReferences();
+
+                    return packages.Select(p => p.AsNuGetReference()).ToList();
```

Comment:
looks like this didn't change but it made me wonder if we should actually force the enumeration here.